### ISSUE TITLE
Add Windows Support

### ISSUE THREAD

**Issue by peterzhuamazon** [opensearch-project/dashboards-maps#55]
Coming from https://github.com/opensearch-project/opensearch-plugins/issues/95, add Windows support.

- [x] Passing CI on Windows
- [ ] Documentation

**Comment by heemin32** [comment]
I don't see the need of documentation for window support as of now.
Please reopen if there is specific area where we need a documentation for window support.

**Comment by heemin32** [comment]
The macos and window support is not backported in 2.x branch.

**Comment by heemin32** [comment]
Another issue is that cypress test is running only for ubuntu now.

**Comment by heemin32** [comment]
Waiting on https://github.com/opensearch-project/opensearch-dashboards-functional-test/issues/359 to get a reference code to support cypress test in window platform.

**Comment by heemin32** [comment]
All other plugins closed the issue without cypress test support in windows platform. Closing it for now as well.

**Comment by dblock** [comment]
Integration/cypress tests need to be running on Windows as part of this, reopening. Please close when there's no gap in testing across platforms, or if something cannot be done, please open a separate issue for it for tracking.

**Comment by peterzhuamazon** [comment]
Hi, just want to make sure this issue is closed, if all the tasks are completed.

Thanks.

**Comment by heemin32** [comment]
OpenSearch is very slow in windows platform during cypress test. It is failing randomly in different steps. After increasing timeout values, the request is failing from backend now.

<img width="1392" alt="Screen Shot 2022-11-23 at 1 38 42 PM" src="https://user-images.githubusercontent.com/1809492/203650414-28bbd057-c6a4-486e-bf74-406c8091b294.png">
